src/nbl/asset/ICPUImage.cpp

```diff
@@ -122,7 +122,7 @@ class CFlattenRegionsStreamHashImageFilter : public CMatchedSizeInOutImageFilter
 
 		auto getScratchAsBuffer = [&memory = state->scratch.memory](size_t size, size_t offset = 0ull)
 		{
-			return core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>, true> >(size, (uint8_t*)memory + offset, core::adopt_memory); // adopt memory & don't free it on exit
+				return ICPUBuffer::create({ size, (uint8_t*)memory + offset }, core::adopt_memory); // adopt memory & don't free it on exit
```

Comment:
this is a wrong substitution, you MUST replace the `core::null_allocator` users with making sure that `std::pmr::null_memory_resource` is being used

https://en.cppreference.com/w/cpp/memory/null_memory_resource